pgraph/pgraph_test.go

```diff
@@ -627,6 +627,45 @@ func TestPgraphT11(t *testing.T) {
 	}
 }
 
+func TestPgraphT12(t *testing.T) {
+	G := NewGraph("g")
+	v1 := NV("v1")
+	v2 := NV("v2")
+	v3 := NV("v3")
+	v4 := NV("v4")
+	v5 := NV("v5")
+	v6 := NV("v6")
+	e1 := NewEdge("e1")
+	e2 := NewEdge("e2")
+	e3 := NewEdge("e3")
+	e4 := NewEdge("e4")
+	e5 := NewEdge("e5")
+	e6 := NewEdge("e6")
+	G.AddEdge(v1, v2, e1)
+	G.AddEdge(v2, v3, e2)
+	G.AddEdge(v3, v4, e3)
+	G.AddEdge(v4, v5, e4)
+	G.AddEdge(v5, v6, e5)
+	G.AddEdge(v1, v6, e6)
+
+	num := G.NumEdges()
+	edges := [6]*Edge{e1, e2, e3, e4, e5, e6}
+
+	for _, edge := range edges {
+		G.DeleteEdge(edge)
+
+		if G.NumEdges() >= num {
```

Comment:
Probably a good idea to make sure `DeleteEdge` isn't adding anything to the structure.

I'll introduce a variable to store the previous `G` to do so.